test/test-component.ts

```diff
@@ -46,10 +46,10 @@ export function createProcessedSnapshotStorageComponent(): IProcessedSnapshotSto
   const processedSnapshots = new Set()
 
   return {
-    async processedFrom(snapshotHashes: string[]) {
+    async filterProcessedSnapshotsFrom(snapshotHashes: string[]) {
       return new Set(snapshotHashes.filter(h => processedSnapshots.has(h)))
     },
-    async saveProcessed(snapshotHash: string) {
+    async saveAsProcessed(snapshotHash: string) {
```

Comment:
markSnapshotAsProcessed may be more precise in this case, since we are not storing the snapshot _itself_, but only its CID/hash
```suggestion
    async saveAsProcessed(snapshotHash: string) {
```

Reply:
changed